[#41530] Copying a project shall also copy file links attached to all work packages

fraber opened the pull request:
https://community.openproject.org/work_packages/41530

Kharonus committed:
[#41530] start and passing over to frank

fraber committed:
Added copying of ::Storages::ProjectStorages to Projects::CopyService

fraber committed:
Added copying of ::Storages::FileLinks to Projects::CopyService

apfohl reviewed (approved):
Besides any side effects I can not know by now it looks good to me.

wielinde commented on config/locales/en.yml:
```diff
@@ -1827,6 +1827,7 @@ en:
   label_project_new: "New project"
   label_project_plural: "Projects"
   label_project_settings: "Project settings"
+  label_project_storage_plural: "Project storages"
```
@fraber Where, in which UI, does that string appear? The term "Project storages" does not make sense for normal users or admins. I argue that this label is not of any help for humans. It is a technical name that stems from the join table naming convention of Rails. A better name could be "Activations of file storages per project"... ok, looks long and clumsy but at least it explains better what it is.

CC @apfohl

fraber replied:
The string appears in /projects/demo-project/copy in the "Copy Options" section. 
It appears together with other options including "Boards", "Forums", "Work package attachments" etc as the options of what objects to copy when creating a copy of a project.

I find it very difficult to come up with a reasonable name. Your "Activations of ..." above clearly doesn't match the stype of the other options, as it is too long.

Saying that, following the style of the other options, a better choice might be just "Storages", as most other options are kept short and "Custom queries" or "Versions" may also not necessarily be understood by users. Maybe you want to add a ticket to add help text to the page to explain? I'll change to "Storages".

wielinde commented on config/locales/en.yml:
```diff
@@ -1978,6 +1979,7 @@ en:
   label_wiki_start: "Start page"
   label_work_package: "Work package"
   label_work_package_attachments: "Work package attachments"
+  label_work_package_file_link_plural: "Work package file links"
```
```suggestion
  label_work_package_file_link_plural: "Links between work packages and files in external file storages"
```
@fraber Here also, "file link" is not a easy to understand concept for normal users/admins. That is why we almost no where use that term, apart from the permissions.

CC @apfohl

fraber replied:
Please see the comment above concerning "Project storages".
Again very difficult to find a good name. So here is my argumentation, please override later if that is important to you.
As with "Project storages" above, "Links between work packags and files in external file storages" does not match the style of other options on that page like "Work package attachments" or "Work package categories".

Also, the list of options could be ordered alphabetically. maybe create a Ticket for it?

fraber committed:
Changed project copy option "Project storages" to "Storages"

A user has multiple options about what sub-objects to copy when
creating a copy of a project in /projects/demo-project/copy.
Copying the "association of a project with a storage" is now
named "Storages", in order to be consistent with the style of the
other items in that list.
